modules/datahandling/userrequests.go

```diff
@@ -82,17 +89,33 @@ func (f *userLoginRequest) setAbstractRequest(req *abstractRequest) {
 }
 
 func (f userLoginRequest) process(db dbfs.DBFS) ([]dhClosure, error) {
-	// TODO (non-immediate/required): implement login logic
+	hashed, err := db.MySQLUserGetPass(f.Username)
+	if err != nil {
+		return []dhClosure{toSenderClosure{msg: newEmptyResponse(fail, f.Tag)}}, err
+	}
 
-	fmt.Printf("Received login request from %s. Login logic not implemented yet.\n", f.Username)
+	if err := bcrypt.CompareHashAndPassword([]byte(hashed), []byte(f.Password)); err != nil {
+		return []dhClosure{toSenderClosure{msg: newEmptyResponse(unauthorized, f.Tag)}}, err
+	}
+
+	token := jwt.NewWithClaims(jwt.SigningMethodES256, tokenPayload{
+		Username:     f.Username,
+		CreationTime: time.Now().Unix(),
+		Validity:     time.Now().Add(1 * time.Hour).Unix(),
```

Comment:
I figured 1 hour would be a short enough window to make sure that even if they deleted their account, the token would expire soon.